File: Ser2023/Company/Company.Web/Modules/ServerTypes/Company/PersonRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib/q";

export interface PersonRow {
    BusinessEntityId?: number;
    Title?: string;
    FirstName?: string;
    MiddleName?: string;
    LastName?: string;
    FullName?: string;
    Suffix?: string;
    IsActive?: number;
    TenantId?: number;
    InsertUserId?: number;
    InsertDate?: string;
    UpdateUserId?: number;
    UpdateDate?: string;
}

export abstract class PersonRow {
    static readonly idProperty = 'BusinessEntityId';
    static readonly isActiveProperty = 'IsActive';
    static readonly nameProperty = 'FullName';
<<<<<<< HEAD
    static readonly localTextPrefix = 'Company.Person';
=======
    static readonly localTextPrefix = 'Person.Person';
>>>>>>> revert
    static readonly lookupKey = 'Company.Person';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<PersonRow>('Company.Person') }
    static async getLookupAsync() { return getLookupAsync<PersonRow>('Company.Person') }

    static readonly deletePermission = 'Company:Person:Delete';
    static readonly insertPermission = 'Company:Person:Modify';
    static readonly readPermission = 'Company:Person:View';
    static readonly updatePermission = 'Company:Person:Modify';

    static readonly Fields = fieldsProxy<PersonRow>();
}

File: Ser2025/Company/Company.Web/Modules/Company/Person/PersonRow.cs
using Company.Administration.Entities;

namespace Company.Company;

[ConnectionKey("Company"), Module("Company"), TableName("[person].[Person]")]
[DisplayName("Person"), InstanceName("Person")]
[ReadPermission(PermissionKeys.Person.View)]
[ModifyPermission(PermissionKeys.Person.Modify)]
[DeletePermission(PermissionKeys.Person.Delete)]
[ServiceLookupPermission("Administration:General")]
public sealed class PersonRow : LoggingRow<PersonRow.RowFields>, IIdRow, INameRow, IIsActiveRow, IMultiTenantRow
{
    const string jBusinessEntity = nameof(jBusinessEntity);

    [DisplayName("Business Entity"), Column("BusinessEntityID"), PrimaryKey, NotNull, ForeignKey(typeof(BusinessEntityRow))]
    [LeftJoin(jBusinessEntity), IdProperty, ServiceLookupEditor(typeof(BusinessEntityRow))]
    public int? BusinessEntityId { get => fields.BusinessEntityId[this]; set => fields.BusinessEntityId[this] = value; }

    [DisplayName("Person Type"), Size(2), NotNull]
    public string PersonType { get => fields.PersonType[this]; set => fields.PersonType[this] = value; }

    [DisplayName("Title"), Size(8)]
    public string Title { get => fields.Title[this]; set => fields.Title[this] = value; }

    [DisplayName("First Name"), Size(50), NotNull]
    public string FirstName { get => fields.FirstName[this]; set => fields.FirstName[this] = value; }

    [DisplayName("Middle Name"), Size(50), NotNull]
    public string MiddleName { get => fields.MiddleName[this]; set => fields.MiddleName[this] = value; }

    [DisplayName("Last Name"), Size(50), NotNull]
    public string LastName { get => fields.LastName[this]; set => fields.LastName[this] = value; }

    [DisplayName("FullName"), NameProperty, QuickSearch]
    [Concat($"T0.[{nameof(FirstName)}]", "' '", $"T0.[{nameof(MiddleName)}]", "' '", $"T0.[{nameof(LastName)}]")]
    public string FullName { get => fields.FullName[this]; set => fields.FullName[this] = value; }

    [DisplayName("Suffix"), Size(10)]
    public string Suffix { get => fields.Suffix[this]; set => fields.Suffix[this] = value; }

    #region Tenant & Activ

    [Insertable(false), Updatable(false)]
    public Int32? TenantId { get => fields.TenantId[this]; set => fields.TenantId[this] = value; }

    public Int32Field TenantIdField { get => fields.TenantId; }

    [NotNull, Insertable(false), Updatable(true)]
    public Int16? IsActive { get => fields.IsActive[this]; set => fields.IsActive[this] = value; }

    Int16Field IIsActiveRow.IsActiveField { get => fields.IsActive; }

    #endregion Tenant & Activ

    public class RowFields : LoggingRowFields
    {
        public Int32Field BusinessEntityId;
        public StringField PersonType;
        public StringField Title;
        public StringField FirstName;
        public StringField MiddleName;
        public StringField LastName;
        public StringField FullName;
        public StringField Suffix;

        public Int16Field IsActive;
        public Int32Field TenantId;

    }
}